File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        set<int> s1 , s2;
        vector<int> num ;

        for(int x : nums1) s1.insert(x);
        for(int y : nums2) s2.insert(y) ;
        
        for(int x : s1){
            if(s2.count(x)){
                num.push_back(x);
            }
        }

        return num ;
    }
};

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cs
public class Solution {
    public int[] Intersection(int[] nums1, int[] nums2) {
        
        HashSet<int> set1 = new  HashSet<int>(nums1);
        HashSet<int> set2 = new  HashSet<int>(nums2);

        List<int> ans = new List<int>(); 

        foreach(int x in set1){
            if(set2.Contains(x)){
                ans.Add(x);
            }
        }

        return ans.ToArray();
    }
}